### PR TITLE
Refactor parameterstore unit test

### PR DESCRIPTION
**Pull request by leewei05**
- refactor `parameterstore_test.go`, I try to follow the style with `secretsmanager_test.go` to keep it consistent
- ignore vim swp files 😁

### CODE REVIEW

File: pkg/provider/aws/parameterstore/parameterstore_test.go [hunk 1]
```diff
@@ -37,216 +37,158 @@ func TestConstructor(t *testing.T) {
...
+	ps := ParameterStore{}
+	for k, v := range successCases {
+		ps.client = v.fakeClient
+		out, err := ps.GetSecret(context.Background(), *v.remoteRef)
+		if !ErrorContains(err, v.expectError) {
+			t.Errorf("[%d] unexpected error: %s, expected: '%s'", k, err.Error(), v.expectError)
 		}
-		if string(out) != row.expectedSecret {
-			t.Errorf("[%d] unexpected secret: expected %s, got %s", i, row.expectedSecret, string(out))
+		if cmp.Equal(out, v.expectedData) {
```
Comment (moolen) [inline pkg/provider/aws/parameterstore/parameterstore_test.go:152]: this should be `v.expectedSecret`. `expectedData` isn't used in this test case and is always `nil`.

File: pkg/provider/aws/parameterstore/parameterstore_test.go [hunk 2]
```diff
@@ -37,216 +37,158 @@ func TestConstructor(t *testing.T) {
...
-	} {
-		f.WithValue(row.apiInput, row.apiOutput, row.apiErr)
-		out, err := p.GetSecret(context.Background(), row.rr)
-		if !ErrorContains(err, row.expectError) {
-			t.Errorf("[%d] unexpected error: %s, expected: '%s'", i, err.Error(), row.expectError)
+
+	// good case: extract property
+	setExtractProperty := func(pstc *parameterstoreTestCase) {
+		pstc.apiOutput.Parameter.Value = aws.String(`{"/shmoo": "bang"}`)
+		pstc.expectedSecret = "bang"
```
Comment (moolen) [inline pkg/provider/aws/parameterstore/parameterstore_test.go:106]: What's missing here is:
```go
pstc.remoteRef.Property = "/shmoo"
```

once this is in and `expectedSecret` is being used below the coverage & functionality should be the same as before!

**Review by moolen — CHANGES_REQUESTED** [review]
Hey @leewei05, sorry it took some time reviewing this one.
Looks good overall thank you for this PR! I found an issue in `TestGetSecret`, see comments below.

**Comment by leewei05** [comment]
@moolen hmm, just fixed the test case, but it seems like the test coverage still couldn't pass. Should I try to rebase this with main branch and try again? 🤔

**Comment by knelasevero** [comment]
> hmm, just fixed the test case, but it seems like the test coverage still couldn't pass. Should I try to rebase this with main branch and try again?

I would not worry that much about the codecov report. Codecov is being a bit agressive on the analysis. We need to make it a bit more forgiving.

**Review by moolen — APPROVED** [review]
LGTM, Thanks you!
Indeed, codecoverage is a bit odd sometimes :/

**Pull request merged**